packages/contracts-bedrock/test/kontrol/pausability-lemmas.md

```diff
@@ -0,0 +1,234 @@
+Kontrol Lemmas
+==============
+
+Lemmas are K rewrite rules that enhance the reasoning power of Kontrol. For more information on lemmas, consult [this section](https://docs.runtimeverification.com/kontrol/guides/advancing-proofs) of the Kontrol documentation.
+
+This file contains the necessary lemmas to run the proofs included in the [proofs](./proofs) folder. Similarly to other files such as [`cheatcodes.md`](https://github.com/runtimeverification/kontrol/blob/master/src/kontrol/kdist/cheatcodes.md) in Kontrol, an idiomatic way of programming in K is with [literate programming](https://en.wikipedia.org/wiki/Literate_programming), allowing to better document the code.
+
+## Imports
+
+For writing the lemmas we use the [`foundry.md`](https://github.com/runtimeverification/kontrol/blob/master/src/kontrol/kdist/foundry.md) file. This file contains and imports all necessary definitions to write the lemmas.
+
+```k
+requires "foundry.md"
+
+module PAUSABILITY-LEMMAS
+    imports BOOL
+    imports FOUNDRY
+    imports INT-SYMBOLIC
+```
+
+Lemmas
+------
+
+## Arithmetic
+
+Lemmas on arithmetic reasoning.
```

Comment:
These general enough lemmas could be part of future versions of Kontrol. Most likely, these lemmas are separate from Kontrol because, in the past, we did not encounter the need to add them.
Needing them in this project is a solid reason to add them to Kontrol, although it is not clear if/when this will be done, hence why we keep them in this project's specific lemmas

Reply:
Ah interesting, thanks 👌